### PR TITLE
Backport master changes to next

### PR DESCRIPTION
**Pull request by slax57**
We will do this regularly to make the conflicts resolution easier once the v5 is released.

### CODE REVIEW

File: docs/InfiniteList.md
Comment (fzaninotto) [inline docs/InfiniteList.md:251]:
```suggestion
    const { isPending, data } = useInfiniteListController();
```

File: docs/InfiniteList.md
Comment (fzaninotto) [inline docs/InfiniteList.md:255]:
```suggestion
                {!isPending && (
```

File: docs/InfiniteList.md
Comment (fzaninotto) [inline docs/InfiniteList.md:224]:
```suggestion
            <WithListContext render={({ isPending, data }) => (
                    !isPending && (
```

File: docs/Features.md
Comment (fzaninotto) [inline docs/Features.md:323]:
```suggestion
const PostList = () => {
  const { data, page, total, setPage, isPending } = useListController({
    sort: { field: 'published_at', order: 'DESC' },
    perPage: 10,
  });
  const handleTableChange = (pagination) => {
    setPage(pagination.current);
  };
  return (
    <>
      <div style={{ margin: 10, textAlign: 'right' }}>
        <Link to="/posts/create">
          <Button icon={<PlusOutlined />}>Create</Button>
        </Link>
      </div>
      <Card bodyStyle={{ padding: '0' }} loading={isPending}>
        <Table
          size="small"
          dataSource={data}
          columns={columns}
          pagination={{ current: page, pageSize: 10, total }}
          onChange={handleTableChange}
        />
      </Card>
    </>
  );
};
```

File: docs/List.md
Comment (fzaninotto) [inline docs/List.md:1205]:
```suggestion
const ProductList = () => (
    <ListBase>
        <Container>
            <Typography variant="h4">All products</Typography>
            <WithListContext render={({ isPending, data }) => (
                    !isPending && (
                        <Stack spacing={1}>
                            {data.map(product => (
                                <Card key={product.id}>
                                    <CardContent>
                                        <Typography>{product.name}</Typography>
                                    </CardContent>
                                </Card>
                            ))}
                        </Stack>
                    )
                )} />
            <WithListContext render={({ isPending, total }) => (
                !isPending && <Typography>{total} results</Typography>
            )} />
        </Container>
    </ListBase>
);
```

File: docs/List.md
Comment (fzaninotto) [inline docs/List.md:1235]:
```suggestion
const ProductList = () => {
    const { isPending, data, total } = useListController();
    return (
        <Container>
            <Typography variant="h4">All products</Typography>
                {!isPending && (
                    <Stack spacing={1}>
                        {data.map(product => (
                            <Card key={product.id}>
                                <CardContent>
                                    <Typography>{product.name}</Typography>
                                </CardContent>
                            </Card>
                        ))}
                    </Stack>
                )}
            {!isPending && <Typography>{total} results</Typography>}
        </Container>
    );
};
```

File: docs/useListController.md
Comment (fzaninotto) [inline docs/useListController.md:48]:
```suggestion
  const { data, page, total, setPage, isPending } = useListController({
    sort: { field: 'published_at', order: 'DESC' },
    perPage: 10,
  });
  const handleTableChange = (pagination) => {
    setPage(pagination.current);
  };
  return (
    <>
      <div style={{ margin: 10, textAlign: 'right' }}>
        <Link to="/posts/create">
          <Button icon={<PlusOutlined />}>Create</Button>
        </Link>
      </div>
      <Card bodyStyle={{ padding: '0' }} loading={isPending}>
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/useListController.md
Comment (slax57) [reply docs/useListController.md:48]: nice catches!

**Comment by slax57** [comment]
We'll wait for #9481 to be merged before merging this one

**Comment by slax57** [comment]
Back to RFR

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**
